http-netty/src/main/java/io/micronaut/http/netty/body/NettyRawJsonStringWriter.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright 2017-2024 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.http.netty.body;
+
+import io.micronaut.context.annotation.Replaces;
+import io.micronaut.core.annotation.Internal;
+import io.micronaut.core.io.buffer.ByteBuffer;
+import io.micronaut.core.type.Argument;
+import io.micronaut.core.type.Headers;
+import io.micronaut.core.type.MutableHeaders;
+import io.micronaut.http.HttpRequest;
+import io.micronaut.http.MediaType;
+import io.micronaut.http.MutableHttpResponse;
+import io.micronaut.http.body.ChunkedMessageBodyReader;
+import io.micronaut.http.body.MessageBodyReader;
+import io.micronaut.http.body.MessageBodyWriter;
+import io.micronaut.http.body.TextPlainBodyHandler;
+import io.micronaut.http.codec.CodecException;
+import io.micronaut.json.body.JsonMessageHandler;
+import io.micronaut.json.body.RawJsonStringHandler;
+import jakarta.inject.Singleton;
+import org.reactivestreams.Publisher;
+
+import java.io.InputStream;
+import java.io.OutputStream;
+
+/**
+ * A JSON body should not be escaped or parsed as a JSON value.
+ *
+ * @author Denis Stepanov
+ * @since 4.6
+ */
+@Singleton
+@Replaces(RawJsonStringHandler.class)
+@JsonMessageHandler.ProducesJson
+@JsonMessageHandler.ConsumesJson
+@Internal
+public final class NettyRawJsonStringWriter implements MessageBodyWriter<CharSequence>, MessageBodyReader<String>, NettyBodyWriter<CharSequence>, ChunkedMessageBodyReader<String> {
```

Comment:
Not sure 100% how did it work before.
Without this the object type JSON writer will pick it up and convert it into the JSON. 
This writer is more correct because we actually want to convert strings as a plain text instead of wrapping it.

Reply:
Raw handlers were internal. Why would you disallow custom handlers?

Reply:
I don't think there is any problem. Body readers / writers are since v4 and if a user chooses to define a custom one it's not our problem. WDYT @graemerocher ?

Reply:
@yawkat @dstepanov  were any tests altered in this PR that introduce a breaking change in behaviour? @yawkat is the behaviour you describe tested somewhere?

Reply:
I believe it is only tested for json, we regularly return `String` from json controllers and expect it to be sent raw. @dstepanov fixed this for json specifically by introducing the NettyRawJsonStringWriter, but that means that now, a typed raw handler is needed for every content type that also declares a normal handler. So imo the existence NettyRawJsonStringWriter is hiding this issue by fixing it for one media type only.

Reply:
Typed handlers are only used for client.
Having a raw string writer makes sense only for text types like XML, JSON. Protocol buffers will need to persist the string not just send it as bytes.
So I would say the behviour now is more correct, default convertion to string shouldn't work for every type.

Reply:
Can argue about whether String should be treated as a raw type all day, but not doing so would be a pretty major breaking change.

Reply:
I can introduce a new writer with a higher order and `@Requires(property = "micronaut.http.legacy-always-string-writer", value = "true")` for people to enable it if needed.

Reply:
why can't we just have the old behavior?

Reply:
Bacase that is esentially broken for writers that don't use textual formats like protocol buffers.

Reply:
that's how it's always been. If you want to change it, do so in 5.0.

Reply:
So actually it was working like that anyway, there is no need for extra handler. Removed unneeded handlers and added tests.

Reply:
thanks!